### DISCUSSION TITLE
Need to uninstall Dhizuku and owndroid

### DISCUSSION THREAD

**Discussion by Johngithubandroidpc** [BinTianqi/OwnDroid#83] [Q&A]
I need to uninstall Dhizuku, not shizuku, it was a typo. Thank you! 

> Uninstall shizuku: 
> ```bash
> pm uninstall moe.shizuku.privileged.api
> ```
> 
> Uninstall OwnDroid, it works only when OwnDroid isn't device owner
> ```bash
> pm uninstall com.bintianqi.owndroid
> ```
> 
> By the way, if you still cannot solve this problem, please create a [discussion](https://github.com/BinTianqi/OwnDroid/discussions), because your question is not related to this issue.

**Comment by BinTianqi** [comment]
Uninstall Dhizuku, it works only when Dhizuku isn't Device owner
```
pm uninstall com.rosan.dhizuku
```
Uninstall OwnDroid, it works only when OwnDroid isn't device owner
```
pm uninstall com.bintianqi.owndroid
```

**Reply by Johngithubandroidpc** [reply]
I'm having difficulty removing Dhizuku's device owner privilege. What is the best way to do this? Thank you.

**Comment by BinTianqi** [comment]
Can you get root privilege in adb shell?

If you can:
```bash
rm /data/system/device_policies.xml
rm /data/system/device_owner_2.xml
reboot
```

**Reply by Johngithubandroidpc** [reply]
Thank you so much, BinTianqi, for patiently answering all my questions! I have a couple more: how can I grant device owner permissions to Tasker?

I also have another phone running Android 7, where I was able to set up device owner mode for both of your apps and apply the DAX patch. However, I can’t install LSpatch on any device below Android 9. Is there a way to get Icebox working without needing LSpatch or connecting to a computer for ADB?

Thanks again!

**Reply by BinTianqi** [reply]
I know nothing about Tasker, please go to it's official website to get help.

You can read IceBox docs here
https://github.com/heruoxin/Ice-Box-Docs/

**Reply by Johngithubandroidpc** [reply]
Hi BinTianqi,

I think you're incredibly talented at what you do, and I really appreciate your straightforward and clear answers—they show just how intelligent you are. I was a bit surprised when you mentioned you weren't familiar with Tasker, but I’d encourage you to take a look at it. If you could explore granting Tasker with device owner permissions, I believe it could bring even bigger world recognition to your work.

**Reply by Johngithubandroidpc** [reply]
> I know nothing about Tasker, please go to it's official website to get help.
> 
> You can read IceBox docs here
> https://github.com/heruoxin/Ice-Box-Docs/

I was able to set up device owner mode for both apps and successfully installed the DEX and Lspatch. I also patched Icebox, but I couldn't use the paid version from the Google Play Store. It seems I would need to pay for this patched version separately. Could you guide me on how to patch it while still using my Google Play Store version of Icebox? Thank you!

**Reply by BinTianqi** [reply]
Do you mean that when you launch paid version of IceBox, it crashed?

**Reply by Johngithubandroidpc** [reply]
When I try to patch the Icebox app, it asks me to uninstall the paid version and reinstall the patched version because the signatures don't match. After launching the patched app, it prompts me to unlock the pro version. Previously, with the paid version from the Google Play Store, I could simply click "Unlock Pro," and it would recognize my purchase automatically. However, with the patched version, clicking "Unlock Pro" redirects me to payment options like Alipay, instead of recognizing my prior purchase through the Google Play Store. This means I have to choose between using the unpatched Google Play version or the patched version, which requires repurchasing the app through a different payment method like Alipay.

**Reply by BinTianqi** [reply]
2 possible reasons:
1. IceBox has an anti-crack system, it checks IceBox apk signature
2. Google play reject you because the patched IceBox's signature doesn't match the signature that IceBox's developer provided to Google.

I can't help you solve this issue. If you can, ask the IceBox's developer

**Reply by Johngithubandroidpc** [reply]
From my experience with LuckyPatcher, it’s the developer’s responsibility to bypass signature verification or fake the signature so that the Google Play Store doesn’t detect any issues. Could you consider dedicating some time to implement this feature in your app, Dhizuku? Thank you!

**Reply by BinTianqi** [reply]
No, signature checking is common in paid apps, because developers don't want user to hook/crack their apps.

**Reply by Johngithubandroidpc** [reply]
> No, signature checking is common in paid apps, because developers don't want user to hook/crack their apps.

I'm excited about your apps because they allow granting device owner privileges to other apps. However, if it's not possible to bypass the signature check from paid app developers or the Play Store, it defeats the purpose of using device owner privileges for most excellent apps, as many of them are paid. In that case, your so-called OwnDroid is of no use to me. While the name sounds promising, it doesn't provide any real benefit for paid apps, which is disappointing.
